### DISCUSSION TITLE
alda repl --server --host  NOT listen on my specific host

### DISCUSSION THREAD

**Discussion by yueyoum** [alda-lang/alda#499] [Q&A]
![alda_server](https://github.com/alda-lang/alda/assets/2081686/bc4d997f-b6d6-43da-a4c6-e84c86d10090)

![alda_client](https://github.com/alda-lang/alda/assets/2081686/44165574-30c8-45aa-949b-8e19d61c4efd)

alda repl --server --host <IP>  still only listen on localhost

**Comment by daveyarwood** [comment]
One thing I should clarify here is that the `--host` option is only used for the client side. On the server side, it's hard-coded to always listen on `localhost`.

We could consider customizing it, or hard-coding it to 0.0.0.0 instead, if that works better. It seems like 0.0.0.0 might be better. Is it not working for you with localhost?

**Reply by yueyoum** [reply]
No problem with localhost.


I trying this due to you post in  #497 

> But you can also run an Alda REPL server and client separate from each other, even on different machines. 
> You can then use alda repl --host 1.2.3.4 --port 9999 --message "..." to send messages to the REPL server.
